packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -125,34 +130,36 @@ moduleFor('Syntax test: {{#with as}}', class extends TogglingSyntaxConditionalsT
   }
 
   ['@test can access alias of an array']() {
-    this.render(`{{#with arrayThing as |thing|}}{{#each thing as |value|}}{{value}}{{/each}}{{/with}}`, {
-      arrayThing: ['a', 'b', 'c', 'd']
+    this.render(`{{#with arrayThing as |words|}}{{#each words as |word|}}{{word}}{{/each}}{{/with}}`, {
+      arrayThing: emberA(['Hello', ' ', 'world'])
     });
 
-    this.assertText('abcd');
+    this.assertText('Hello world');
 
     this.runTask(() => this.rerender());
 
-    this.assertText('abcd');
-  }
+    this.assertText('Hello world');
 
-  ['@test empty arrays yield inverse']() {
-    this.render(`{{#with arrayThing as |thing|}}{{thing}}{{else}}Empty Array{{/with}}`, {
-      arrayThing: []
+    this.runTask(() => {
+      let array = get(this.context, 'arrayThing');
+      array.replace(0, 1, 'Goodbye');
+      array.removeAt(1);
+      array.insertAt(1, ', ');
+      array.pushObject('!');
     });
 
-    this.assertText('Empty Array');
+    this.assertText('Goodbye, world!');
 
-    this.runTask(() => this.rerender());
+    this.runTask(() => set(this.context, 'arrayThing', ['Hello', ' ', 'world']));
 
-    this.assertText('Empty Array');
+    this.assertText('Hello world');
   }
```

Comment:
"can access alias of an array" is missing mutation/replacement cycle; "empty arrays yield inverse" is already covered in the basic test cases